### PR TITLE
feat: add indicator list type list view

### PR DESCRIPTION
**Pull request by Chisomchima**
Implements [DHIS2-18283](https://dhis2.atlassian.net/browse/DHIS2-18283) and [DHIS2-18284](https://dhis2.atlassian.net/browse/DHIS2-18284)
---

### Description
This PR displays the Indicator type list as well as the Edit and new forms.

---

### Checklist

-   [ ] Indicator Type List View
-   [ ] Create New Indicator type list
-   [ ] Edit Single Indicator type list

---

[DHIS2-18283]: https://dhis2.atlassian.net/browse/DHIS2-18283?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DHIS2-18284]: https://dhis2.atlassian.net/browse/DHIS2-18284?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/indicatorTypes/Edit.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+    const IndicatorTypeQuery = useQuery({
+        queryKey: [query],
+        queryFn: queryFn<IndicatorTypesFormValues>,
+    })
+    return (
+        <FormBase
+            onSubmit={useOnSubmitEdit({ section, modelId })}
+            section={section}
+            initialValues={IndicatorTypeQuery.data as IndicatorTypesFormValues}
```
Comment (Birkbjo) [inline src/pages/indicatorTypes/Edit.tsx:39]: This should be 
```suggestion
            initialValues={IndicatorTypeQuery.data}
```
should be inferred from `useQuery`

File: src/pages/indicatorTypes/Edit.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    const modelId = useParams().id as string
+
+    const query = {
+        resource: 'indicatorTypes',
+        id: modelId,
+        params: {
+            fields: fieldFilters.concat(),
+        },
+    }
+    const IndicatorTypeQuery = useQuery({
```
Comment (Birkbjo) [inline src/pages/indicatorTypes/Edit.tsx:31]: Minor, but I shouldn't be capitalized here. 
```suggestion
    const indicatorTypeQuery = useQuery({
```

File: src/components/form/FormBase.tsx [hunk 3]
```diff
@@ -10,6 +10,8 @@ import { useCustomAttributesQuery } from './attributes'
 
 type MaybeModelWithAttributes = {
     id?: string
+    name?: string
+    factor?: string
```
Comment (Birkbjo) [inline src/components/form/FormBase.tsx:14]: I don't think we need to add `factor` here?

File: src/components/form/FormBase.tsx [hunk 4]
```diff
@@ -44,9 +46,9 @@ export function FormBase<TInitialValues extends MaybeModelWithAttributes>({
         }
     }, [customAttributes.data, initialValues, includeAttributes])
 
-    if (customAttributes.error) {
-        return <NoticeBox error title="Failed to load custom attributes" />
-    }
+    // if (customAttributes.error) {
+    //     return <NoticeBox error title="Failed to load custom attributes" />
+    // }
```
Comment (Birkbjo) [inline src/components/form/FormBase.tsx:51]: Remove?

File: src/types/generated/models.ts [hunk 5]
```diff
@@ -4776,7 +4776,7 @@ export type IndicatorType = {
     created: string
     createdBy: User
     displayName: string
-    factor: number
+    factor: string
```
Comment (Birkbjo) [inline src/types/generated/models.ts:4779]: This is actually a number from the API. So please don't change this. Ref: https://debug.dhis2.org/dev/api/schemas/indicatorType (search for `factor`). Or eg. https://debug.dhis2.org/dev/api/indicatorTypes?fields=displayName,id,factor .
These type reflect the API, and we should instead convert the values in the field.

File: src/pages/indicatorTypes/form/IndicatorTypesSchema.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { z } from 'zod'
+import { getDefaults, createFormValidate, modelFormSchemas } from '../../../lib'
+
+const { identifiable } = modelFormSchemas
+
+export const IndicatorSchema = identifiable.extend({
+    name: z.string().min(1, 'Name is required'),
+    factor: z
```
Comment (Birkbjo) [inline src/pages/indicatorTypes/form/IndicatorTypesSchema.ts:8]: I think the best way to handle `factor` being an integer when fetching, while also support string is to coerce it. Eg. 

```
factor: z.coerce.string().min(1, 'Factor is required')
```

File: src/components/form/FormBase.tsx [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import { NoticeBox } from '@dhis2/ui'
+// import { NoticeBox } from '@dhis2/ui'
```
Comment (Birkbjo) [inline src/components/form/FormBase.tsx:1]: Remove?

File: src/pages/indicatorTypes/form/IndicatorTypesSchema.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { z } from 'zod'
+import { getDefaults, createFormValidate, modelFormSchemas } from '../../../lib'
+
+const { identifiable } = modelFormSchemas
+
+export const IndicatorSchema = identifiable.extend({
+    name: z.string().min(1, 'Name is required'),
```
Comment (Birkbjo) [inline src/pages/indicatorTypes/form/IndicatorTypesSchema.ts:7]: `identifiable`-schema should already check for required?

File: src/pages/indicatorTypes/form/IndicatorTypesFormFields.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+                    {i18n.t('Basic information')}
+                </StandardFormSectionTitle>
+                <StandardFormSectionDescription>
+                    {i18n.t(
+                        'Set up the basic information for this Indicator Type.'
+                    )}
+                </StandardFormSectionDescription>
+
+                <StandardFormField>
+                    <Field
```
Comment (Birkbjo) [inline src/pages/indicatorTypes/form/IndicatorTypesFormFields.tsx:27]: You could also use the re-usable `NameField` component - which has some additional validations, like checking if the value is taken.

File: src/components/form/FormBase.tsx [hunk 10]
```diff
@@ -44,10 +44,6 @@ export function FormBase<TInitialValues extends MaybeModelWithAttributes>({
         }
     }, [customAttributes.data, initialValues, includeAttributes])
 
-    if (customAttributes.error) {
```
Comment (Birkbjo) [inline src/components/form/FormBase.tsx:47]: Why is this removed?

File: src/lib/sectionList/listViews/sectionListViewsConfig.ts [hunk 11]
```diff
@@ -128,12 +128,30 @@ export const modelListViewsConfig = {
...
         filters: {
             default: ['indicatorType'],
         },
     },
+    indicatorType: {
+        columns: {
+            default: [
+                'name',
+                { label: i18n.t('Factor'), path: 'factor' },
+                DESCRIPTORS.publicAccess,
```
[the remark this answers is not in the document]
Comment (Chisomchima) [reply src/lib/sectionList/listViews/sectionListViewsConfig.ts:147]: Good catch Birk, will be removed in the next commit.

File: src/pages/indicatorTypes/form/IndicatorTypesFormFields.tsx (hunk 9 above)
Comment (Chisomchima) [reply src/pages/indicatorTypes/form/IndicatorTypesFormFields.tsx:27]: I am adding this to the next commit.

File: src/lib/sectionList/listViews/sectionListViewsConfig.ts [hunk 12]
```diff
@@ -154,12 +154,29 @@ export const modelListViewsConfig = {
...
         },
     },
+    indicatorType: {
+        columns: {
+            default: [
+                'name',
+                { label: i18n.t('Factor'), path: 'factor' },
+                'lastUpdated',
+            ],
+        },
```
Comment (Birkbjo) [inline src/lib/sectionList/listViews/sectionListViewsConfig.ts:175]: We need to remove `publicAccess` from available columns as well. Or the user may select it as part of the view. 
I might come up with a better way to share the non-shareable columns but for now we could just do do:

```
```suggestion
        columns: {
            default: [
                'name',
                { label: i18n.t('Factor'), path: 'factor' },
                'lastUpdated',
            ],
             available: [
                'code',
                'created',
                'createdBy',
                'href',
                'id',
                'lastUpdatedBy',
                ],
        },

File: src/components/form/FormBase.tsx (hunk 10 above)
Comment (Birkbjo) [reply src/components/form/FormBase.tsx:51]: Ah, I think I understand now - indicator type dont have attributes. Should be enough to pass `includeAttributes={false}` to the BaseForm component.

File: src/pages/indicatorTypes/Edit.tsx [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+        queryKey: [query],
+        queryFn: queryFn<IndicatorTypesFormValues>,
+    })
+    return (
+        <FormBase
+            onSubmit={useOnSubmitEdit({ section, modelId })}
+            section={section}
+            initialValues={indicatorTypeQuery.data}
+            validate={validate}
+        >
```
Comment (Birkbjo) [inline src/pages/indicatorTypes/Edit.tsx:41]:
```suggestion
          includeAttributes={false},
        >
```

File: src/components/form/FormBase.tsx [hunk 14]
```diff
@@ -48,10 +48,6 @@ export function FormBase<TInitialValues extends MaybeModelWithAttributes>({
...
-        return <NoticeBox error title="Failed to load custom attributes" />
```
Comment (Birkbjo) [inline src/components/form/FormBase.tsx:52]: I still think we should keep this error though

**Review by Birkbjo — APPROVED** [review]
